hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/encryption.md

```diff
@@ -704,10 +704,20 @@ clients where S3-CSE has not been enabled.
 
 ### Features
 
-- Supports client side encryption with keys managed in AWS KMS.
+- Supports client side encryption with keys managed in AWS KMS (CSE-KMS)
+- Supports client side encryption with custom keys by implementing custom [Keyring](https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/choose-keyring.html) (CSE-CUSTOM)
+- Backward compatible with older encryption clients like `AmazonS3EncryptionClient.java`(V1) and `AmazonS3EncryptionClientV2.java`(V2)
 - encryption settings propagated into jobs through any issued delegation tokens.
 - encryption information stored as headers in the uploaded object.
 
+### Compatibility Issues
+
+- The V1 and V2 clients support reading unencrypted S3 objects, whereas the V3 client does not. In order to read S3 objects in a directory with a mix of encrypted and unencrypted objects.
+- Unlike the V2 and V3 clients which always pads 16 bytes, V1 client pads extra bytes to the next multiple of 16. For example if unencrypted object size is 12bytes, V1 client pads extra 4bytes to make it multiple of 16.
+- The V1 client supports storing encryption metadata in instruction file.
```

Comment:
explain this in its own section and consequences: file rename loses this, file delete doesn't clean it up

Reply:
ack

Reply:
We won't be considering instruction file and documented the same.